lib/pubsub/subscription.js

```diff
@@ -280,15 +280,19 @@ Subscription.prototype.delete = function(callback) {
  * Pull messages from the subscribed topic. If messages were found, your
  * callback is executed with the message object.
  *
+ * @todo Should not be racing with other pull.
```

Comment:
We should open an issue for this with specifics of how to reproduce. I'm not sure the specifics of this TODO.